src/actions/challenges.js

```diff
@@ -169,19 +186,39 @@ export function loadChallengeDetails (projectId, challengeId) {
     if (challengeId) {
       return dispatch({
         type: LOAD_CHALLENGE_DETAILS,
-        payload: fetchChallenge(challengeId).then((challenge) => {
+        payload: fetchChallenge(challengeId).then(challenge => {
           // TODO remove this unncessary check, or better utilize the the case when given project id
           // does not match with challenge's project id
-          if (challenge.projectId == projectId) { // eslint-disable-line
+          if (challenge.projectId === projectId) {
+            // eslint-disable-line
             dispatch(loadProject(projectId))
           }
+          dispatch(loadGroupDetails(challenge.groups))
           return challenge
         })
       })
     }
   }
 }
 
+/**
+ * Loads group details
+ */
+export function loadGroupDetails (groupIds) {
+  return (dispatch, getState) => {
+    const promiseAll = []
+    groupIds.forEach(id => {
+      promiseAll.push(fetchGroupDetail(id))
+    })
```

Comment:
I think that it could be simplified without losing clarity:
```suggestion
    const promiseAll = groupIds.map(id => fetchGroupDetail(id));
```